authorization_services/topics/enforcer-claim-information-point.adoc

```diff
@@ -0,0 +1,165 @@
+              "claim-a": "/a",
+              "claim-d": "/d",
+              "claim-d0": "/d/0",
+              "claim-d-all": ["/d/0", "/d/1"]
+            },
+            "url": "http://mycompany/claim-provider",
+            "method": "POST",
+            "headers": {
+              "Content-Type": "application/x-www-form-urlencoded",
+              "header-b": ["header-b-value1", "header-b-value2"],
+              "Authorization": "Bearer {keycloak.access_token}"
+            },
+            "parameters": {
+              "param-a": ["param-a-value1", "param-a-value2"],
+              "param-subject": "{keycloak.access_token['/sub']}",
+              "param-user-name": "{keycloak.access_token['/preferred_username']}",
+              "param-other-claims": "{keycloak.access_token['/custom_claim']}"
+            }
+          }
+        }
+      }
+    ]
+  }
+```
+
+== Static Claims
+
+.keycloak.json
+```json
+"policy-enforcer": {
+    "paths": [
+      {
+        "path": "/protected/resource",
+        "claim-information-point": {
+          "claims": {
+            "claim-from-static-value": "static value",
+            "claim-from-multiple-static-value": ["static", "value"],
+          }
+        }
+      }
+    ]
+  }
+```
+
+== Claim Information Provider SPI
+
+The Claim Information Provider SPI can be used by developers to support different claim information points in case none of the
+built-ins providers are enough to address their requirements.
+
+For example, to implement a new CIP provider you need to implement `org.keycloak.adapters.authorization.ClaimInformationPointProviderFactory`
+and `ClaimInformationPointProvider` and also provide the file `META-INF/services/org.keycloak.adapters.authorization.ClaimInformationPointProviderFactory`
+in your application`s classpath.
+
+Example of `org.keycloak.adapters.authorization.ClaimInformationPointProviderFactory`:
+
+```java
+public class MyClaimInformationPointProviderFactory implements ClaimInformationPointProviderFactory<MyClaimInformationPointProvider> {
+
+    @Override
+    public String getName() {
+        return "my-claims";
+    }
+
+    @Override
+    public void init(PolicyEnforcer policyEnforcer) {
+
+    }
+
+    @Override
+    public MyClaimInformationPointProvider create(Map<String, Object> config) {
+        return new MyClaimInformationPointProvider(config);
+    }
+}
+```
+
+Every CIP provider must be associated with a name, as defined above in the `MyClaimInformationPointProviderFactory.getName` method. The name
+will be used to map the configuration from the `claim-information-point` section in the `policy-enforcer` configuration to the implementation.
+
+When processing requests, the policy enforcer will call the `MyClaimInformationPointProviderFactory.create` method in order to obtain an instance
+of `MyClaimInformationPointProvider`. At this moment, any configuration defined for this particular CIP provider (via `claim-information-point`) is passed
+as a map.
+
```

Comment:
When processing requests, the policy enforcer will call the `MyClaimInformationPointProviderFactory.create` method in order to obtain an instance of `MyClaimInformationPointProvider`. When called, any configuration defined for this particular CIP provider (via `claim-information-point`) is passed as a map.